### PR TITLE
FE_Nothing: pass reference-cell type to constructor

### PR DESCRIPTION
**Pull request by peterrum**
Needed by #11401.

### CODE REVIEW

File: include/deal.II/fe/fe_nothing.h [hunk 1]
```diff
@@ -91,6 +91,13 @@ class FE_Nothing : public FiniteElement<dim, spacedim>
    */
   FE_Nothing(const unsigned int n_components = 1, const bool dominate = false);
 
+  /**
+   * Same as above but with the possibility to specify a reference-cell type.
```
Comment (Rombur) [inline include/deal.II/fe/fe_nothing.h:95]: You don't have the possibility to specify a reference-cell type, you have to specify it. Right?

File: source/fe/fe_nothing.cc [hunk 2]
```diff
@@ -41,6 +43,17 @@ FE_Nothing<dim, spacedim>::FE_Nothing(const unsigned int n_components,
 }
 
 
+
+template <int dim, int spacedim>
+FE_Nothing<dim, spacedim>::FE_Nothing(const unsigned int n_components,
+                                      const bool         dominate)
+  : FE_Nothing<dim, spacedim>(ReferenceCell::get_hypercube(dim),
```
Comment (Rombur) [inline source/fe/fe_nothing.cc:50]: So by default, we assume the cell type is square or hex. Should we document this?

**Review by Rombur — CHANGES_REQUESTED** [review]

File: source/fe/fe_nothing.cc (hunk 2 above)
Comment (peterrum) [reply source/fe/fe_nothing.cc:50]: I have updated the documentation.

**Review by Rombur — APPROVED** [review]
Looks good but the CI is failing.

**Review by drwells — APPROVED** [review]
This is fine as-is but I can think of some improvements (like updating `get_name()`) - since this is blocking another PR and those are secondary I'll merge this now and write those improvements in a followup.

**Pull request merged**